airbyte-cdk/python/pyproject.toml

```diff
@@ -35,7 +35,7 @@ Jinja2 = "~3.1.2"
 jsonref = "~0.2"
 jsonschema = "~3.2.0"
 pendulum = "<3.0.0"
-pydantic = "^1.10.8"
+pydantic = "^2.7"
```

Comment:
There are a lot of connectors using Pydantic to set up specs and configs. This will force the upgrade on those connectors as well. Is there any change to do for these?

Reply:
Correct, those connectors will have to either fully migrate, or they will have to `from pydantic.v1 import BaseModel`. The latter is cheap.

Reply:
How do we do the latter? I didn't think it was (easily) possible in Python to have two versions of the same lib at the same time

Reply:
Pydantic included a backwards compatibile "v1" API within Pydantic 2.x. 

This allows deferred refactoring. More info on that here:

https://docs.pydantic.dev/latest/migration/#continue-using-pydantic-v1-features

It is a shame that Pydantic never backported this syntax though, because "v1" users ironically cannot use this syntax until they are on v2:

```
# This only works in v2:
from pydantic.v1 import BaseModel
```

Meaning, there's no version of the code that easily supports both.

I got this trick from Langchain, for libraries that need to allow consumers to use either one. Then, whenever our library needs Pydantic imports, it imports from a module containing this definition, instead of importing directly from `pydantic` itself. 

```
# We use Pydantic 1.x for backwards compatibility. We import it in a
# way that is also forwards compatible with Pydantic 2.x.
try:
    import pydantic.v1 import BaseModel as BaseModel_v1
    # If this succeeds we are using Pydantic 2.x, but we import the backwards-compatible
    # version of BaseModel.
except ImportError:
    # In this case, we are using Pydantic 1.x and we import as usual.
    import pydantic import BaseModel as BaseModel_v1
```

@erohmensing - I see you have alternative implementations pending on `airbyte-protocol`. My WIP PR from a while back is here, in case helpful:

- https://github.com/airbytehq/airbyte-protocol/pull/77